### PR TITLE
Chgbomtype2

### PR DESCRIPTION
**Pull request by andrewfraser73**
UI and a bit more backend for bom type change

we have lots of validation checks on the server side trying to replicate most of that also on client

### CODE REVIEW

File: src/Facade/Services/PartFacadeService.cs [hunk 1]
```diff
@@ -64,6 +64,21 @@ public string GetPartNumberFromId(int id)
             }
         }
 
+        public IResult<BomTypeChangeResource> GetBomType(int partNumberId, IEnumerable<string> privileges = null)
```
Comment (richardiphillips) [inline src/Facade/Services/PartFacadeService.cs:67]: can't you just get this information from the part without having a separate facade method for it?

File: src/Service.Host/client/src/middleware/itemCreated.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import * as actionTypes from '../actions/index';
 export default () => next => action => {
     const result = next(action);
 
-    if (action.type.startsWith('RECEIVE_NEW_')) {
+    if (action.type?.startsWith('RECEIVE_NEW_')) {
```
Comment (richardiphillips) [inline src/Service.Host/client/src/middleware/itemCreated.js:8]: i'd revert this. actions should always have a type

File: src/Service.Host/client/src/components/BomTypeChange.js [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+import ModeEditIcon from '@mui/icons-material/ModeEdit';
+import EditOffIcon from '@mui/icons-material/EditOff';
+import Tooltip from '@mui/material/Tooltip';
+import history from '../history';
+import config from '../config';
+import partsActions from '../actions/partsActions';
+import bomTypeChangeActions from '../actions/bomTypeChangeActions';
+import partSuppliersActions from '../actions/partSuppliersActions';
+
+function BomTypeChange() {
```
Comment (richardiphillips) [inline src/Service.Host/client/src/components/BomTypeChange.js:28]: is there a test for this?

**Review by richardiphillips — APPROVED** [review]

File: src/Facade/Services/PartFacadeService.cs (hunk 1 above)
Comment (andrewfraser73) [reply src/Facade/Services/PartFacadeService.cs:67]: I think it could be a possible refactor unless i remember what pushed me down this route

**Pull request merged**